### ISSUE TITLE
Attach a history stack to the session for certain navigation actions

### ISSUE THREAD

**Issue by dguhl** [firefly-iii/firefly-iii#2717]
**Bug description**
I am running Firefly III version 4.7.17.3, and I confirmed that the `return_to_` checkboxes in various creation and edit forms do not react in a logical way.

__Expected__:
When activated, I would return to the edit form for the same record, or to the create form for a new record respectively. When inactive, I would return to the view from which I came to the form, usually the overview form.

__Actual__:
The checkbox does not seem to have any effect. I return to the forms either way.

**Steps to reproduce**
Reproduction would be described on rules.

1. Edit an existing rule.
2. Leave out the last checkbox "Return here after update"
3. Press "Update Rule"

**Extra info**
Brave Version 0.69.132 Chromium: 77.0.3865.90 (Official Build) (64-bit) on Windows 10

_**This bug has been tested and could not be confirmed for dev.**_

I opened this bug report to discuss an improvement on a call history attached to the session and triggered by the router, instead of calling `getPreviousUri` and `rememberPreviousUri` from `UserNavigation`, which, if inconsistently implemented throughout the rest of the application, would lead to bugs like reported above.

**Comment by JC5** [comment]
Ah, sorry, now I see in which version you are. This has been a bug in a few previous versions and has since been fixed. I'm not sure which version, but definitely after this one.

**Comment by dguhl** [comment]
I believe the grounding issue has been fixed in #2490 in Version 4.8.1. The only extension Laravel allows for would be an event listener which is called on every route call on any route method. The current solution in Firefly III seems more efficient to me, because it only acts on occasion instead of storing the route in memory on any routing event. A solution for a general routing history should be separately loadable a Laravel library. 
I opened this bug because I believed it would still exist or would be able to provide a better solution than currently implemented, but two of the reasons I like to contribute to this project is its appealing premise and superior code design, the latter once more proven by examining this issue. If everyone agrees, I'd like to close this issue for now, so we can focus on other things.
